datasafe-business-tests/datasafe-business-tests-random-actions/src/test/java/de/adorsys/datasafe/business/impl/e2e/randomactions/RandomActionsOnSimpleDatasafeAdapterTest.java

```diff
@@ -0,0 +1,225 @@
+
+                    @Override
+                    public InputStream read(ReadRequest<UserIDAuth, PrivateResource> request) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+
+                    @Override
+                    public void remove(RemoveRequest<UserIDAuth, PrivateResource> request) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+
+                    @Override
+                    public OutputStream write(WriteRequest<Set<UserID>, PublicResource> request) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+                };
+            }
+
+            @Override
+            public ProfileOperations userProfile() {
+                return new ProfileOperations() {
+                    @Override
+                    public void registerPublic(CreateUserPublicProfile profile) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+
+                    @Override
+                    public void registerPrivate(CreateUserPrivateProfile profile) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+
+                    @Override
+                    public void registerUsingDefaults(UserIDAuth user) {
+                        datasafeService.createUser(user);
+                    }
+
+                    @Override
+                    public void deregister(UserIDAuth userID) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+
+                    @Override
+                    public UserPublicProfile publicProfile(UserID ofUser) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+
+                    @Override
+                    public UserPrivateProfile privateProfile(UserIDAuth ofUser) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+
+                    @Override
+                    public boolean userExists(UserID ofUser) {
+                        throw new IllegalStateException("Not implemented");
+                    }
+                };
+            }
+        };
+    }
+
+    private DocumentDirectoryFQN asFqnDir(PrivateResource resource) {
+        return new DocumentDirectoryFQN(resource.location().getPath());
+    }
+
+    private DocumentFQN asFqnDoc(PrivateResource resource) {
+        return new DocumentFQN(resource.location().getPath());
+    }
+
+    private ResolvedResource asResolved(Uri root, DocumentFQN resource) {
+        return new BaseResolvedResource(
+                new BasePrivateResource(root, new Uri(""), new Uri(resource.getDatasafePath())),
+                Instant.now()
+        );
+    }
+
+    private DFSCredentials getCredentials(StorageDescriptor descriptor) {
```

Comment:
so this method is a copy of SimpleDatasafeAdapterTest.myinit(). Might be better in a function that can be used by different tests.